lxc/console.go

```diff
@@ -0,0 +1,197 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"strconv"
+
+	"github.com/gorilla/websocket"
+
+	"github.com/lxc/lxd/client"
+	"github.com/lxc/lxd/lxc/config"
+	"github.com/lxc/lxd/shared/api"
+	"github.com/lxc/lxd/shared/gnuflag"
+	"github.com/lxc/lxd/shared/i18n"
+	"github.com/lxc/lxd/shared/logger"
+	"github.com/lxc/lxd/shared/termios"
+)
+
+type consoleCmd struct {
+	showLog bool
+}
+
+func (c *consoleCmd) showByDefault() bool {
+	return true
+}
+
+func (c *consoleCmd) usage() string {
+	return i18n.G(
+		`Usage: lxc console [<remote>:]<container> [-l]
+
+Interact with the container's console device and log.`)
+}
+
+func (c *consoleCmd) flags() {
+	gnuflag.BoolVar(&c.showLog, "show-log", false, i18n.G("Retrieve the container's console log"))
+}
+
+func (c *consoleCmd) sendTermSize(control *websocket.Conn) error {
+	width, height, err := termios.GetSize(int(os.Stdout.Fd()))
+	if err != nil {
+		return err
+	}
+
+	logger.Debugf("Window size is now: %dx%d", width, height)
+
+	w, err := control.NextWriter(websocket.TextMessage)
+	if err != nil {
+		return err
+	}
+
+	msg := api.ContainerExecControl{}
+	msg.Command = "window-resize"
+	msg.Args = make(map[string]string)
+	msg.Args["width"] = strconv.Itoa(width)
+	msg.Args["height"] = strconv.Itoa(height)
+
+	buf, err := json.Marshal(msg)
+	if err != nil {
+		return err
+	}
+	_, err = w.Write(buf)
+
+	w.Close()
+	return err
+}
+
+type ReadWriteCloser struct {
+	io.Reader
+	io.WriteCloser
+}
+
+type StdinMirror struct {
+	r                 io.Reader
+	consoleDisconnect chan<- bool
+}
+
+var foundEscape = false
```

Comment:
can't this be a property of StdinMirror rather than a global variable?

Reply:
If we make it a pointer it should work.

Reply:
The boolean that is. The struct can't be.

Reply:
done